libraries/resource_providers/api200/power_device_provider.rb

```diff
@@ -23,14 +23,53 @@ def discover
         end
       end
 
+      def load_power_device
+        return @item if @item.retrieve!
+        power_devices_list = resource_named(:PowerDevice).get_ipdu_devices(@item.client, @name)
+        power_devices_list.first
+      end
+
       def remove
         # First try to remove by name, if it does not work it consider the power device is an iPDU
         return true if super
-        power_devices_list = resource_named(:PowerDevice).get_ipdu_devices(@item.client, @name)
-        return false if power_devices_list.empty?
-        @item = power_devices_list.first
+        item = load_power_device
+        return false if item.nil?
+        @item = item
         super
       end
+
+      def refresh
```

Comment:
You could use the refresh Mixin, don't you?

Reply:
I thought about this, but this `refresh` call the method `set_refresh_state` that use a hash of options as argument (the string of state is into the options), and it is the unique case of this method set_refresh_state with this parameter. =(
The other reason is when we retrieve the resource item, we made that in different way, because the item can be a IPDU. 
So for this reasons, I thought be better create a new method into power_device instead of change the `refresh` to a particular case.

I thought too using the property `refresh_state` of power_device resource to receive a hash with options, so I could call the refresh of mixin (adjusting it to call some method to retrieve the resource item, because in case of "powser device" can be IPDU and it is retrieved in differente way), but I imaginated that would be wrong semantic of property `refresh_state`, once that would receive the other options like username, password.

What do you think?

Reply:
Ok, no problem